kotlin-native/backend.native/tests/objcexport/coroutines.swift

```diff
@@ -268,34 +288,52 @@ private func testBridges() throws {
 }
 
 private func testImplicitThrows1() throws {
-    var result: KotlinUnit? = nil
     var error: Error? = nil
     var completionCalled = 0
 
+#if LEGACY_SUSPEND_UNIT_FUNCTION_EXPORT
+    var result: KotlinUnit? = nil
+
     CoroutinesKt.throwCancellationException { _result, _error in
         completionCalled += 1
         result = _result
         error = _error
     }
 
-    try assertEquals(actual: completionCalled, expected: 1)
     try assertNil(result)
+#else
+    CoroutinesKt.throwCancellationException { _error in
+        completionCalled += 1
+        error = _error
+    }
+#endif
+
+    try assertEquals(actual: completionCalled, expected: 1)
     try assertTrue(error?.kotlinException is KotlinCancellationException)
 }
 
 private func testImplicitThrows2() throws {
-    var result: KotlinUnit? = nil
     var error: Error? = nil
     var completionCalled = 0
 
+#if LEGACY_SUSPEND_UNIT_FUNCTION_EXPORT
+    var result: KotlinUnit? = nil
+
     ThrowCancellationExceptionImpl().throwCancellationException { _result, _error in
+            completionCalled += 1
+            result = _result
+            error = _error
+    }
+
+    try assertNil(result)
+#else
+    ThrowCancellationExceptionImpl().throwCancellationException { _error in
         completionCalled += 1
-        result = _result
         error = _error
     }
+#endif
```

Comment:
(I noticed this only now but anyway) a test checking the non-throwing call with "unit completion" from Swift to Kotlin seems missing. Could you please add it?

Reply:
Added.

This is probably outside of the scope here, but would it make sense to also have tests invoking suspend functions as async functions from Swift?

Reply:
Thanks!
Testing Kotlin `suspend` as Swift `async` would be great, but our test infrastructure is not exactly ready for this at the moment. In particular, testing this on macOS requires macOS 12.